hydra-node/src/Hydra/Node.hs

```diff
@@ -214,11 +215,13 @@ processNextEvent ::
   STM m (Outcome tx)
 processNextEvent HydraNode{nodeState, ledger, env} e =
   modifyHeadState $ \s ->
-    let outcome = Logic.update env ledger s e
+    let outcome = computeOutcome s e
      in (outcome, aggregateState s outcome)
  where
   NodeState{modifyHeadState} = nodeState
 
+  computeOutcome = Heads.update (Logic.update env ledger)
```

Comment:
I like this way of composing logic functions!

We could compose multiple of such in a chain and have a final `\st ev -> Error $ InvalidEvent ev st` component (where the error should be named "Unhandled event" or so)